src/asound/bluealsa-ctl.c

```diff
@@ -992,23 +1120,90 @@ SND_CTL_PLUGIN_DEFINE_FUNC(bluealsa) {
 		goto fail;
 	}
 
-	if (!bluealsa_dbus_get_pcms(&ctl->dbus_ctx, &ctl->pcm_list, &ctl->pcm_list_size, &err)) {
+	if (!bluealsa_dbus_get_pcms(&ctl->dbus_ctx, &pcm_list, &pcm_list_size, &err)) {
 		SNDERR("Couldn't get BlueALSA PCM list: %s", err.message);
 		ret = -ENODEV;
 		goto fail;
 	}
 
+	if (bacmp(&device, BDADDR_ALL) != 0) {
+		struct ba_pcm *_pcms = NULL;
+		size_t n, count = 0;
+
+		ctl->single_device = true;
+
+		if (bacmp(&device, BDADDR_ANY) == 0) {
+			/* Interpret BDADDR_ANY as a request for the most
+			 * recently connected bluetooth audio device */
+			uint32_t seq = 0;
+			size_t latest = 0;
+
+			if (pcm_list_size == 0) {
+				SNDERR("No BlueALSA audio devices connected");
+				ret = -ENODEV;
+				goto fail;
+			}
+
+			for (n = 0; n < pcm_list_size; n++) {
+				if (pcm_list[n].sequence >= seq) {
+					seq = pcm_list[n].sequence;
+					latest = n;
+				}
+			}
+
+			bacpy(&device, &pcm_list[latest].addr);
+		}
+
+		for (n = 0; n < pcm_list_size; n++) {
```

Comment:
Add a comment that here we are filtering PCMs which belong to selected BT device, or something like that.